File: common/src/main/java/org/sandboxpowered/silica/content/Colour.java
package org.sandboxpowered.silica.content;

import java.util.Locale;

public enum Colour {
    WHITE,
    ORANGE,
    MAGENTA,
    LIGHT_BLUE,
    YELLOW,
    LIME,
    PINK,
    GRAY,
    LIGHT_GRAY,
    CYAN,
    PURPLE,
    DARK_BLUE,
    BROWN,
    GREEN,
    RED,
    BLACK;

    private String name;

    Colour() {
        this.name = name().toLowerCase(Locale.ENGLISH);
    }

    Colour(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/util/content/Colour.kt
package org.sandboxpowered.silica.util.content

import org.sandboxpowered.silica.state.property.StringSerializable

enum class Colour(override val asString: String) : StringSerializable {
    BLACK("black"),
    RED("red"),
    GREEN("green"),
    BROWN("brown"),
    BLUE("blue"),
    PURPLE("purple"),
    CYAN("cyan"),
    LIGHT_GRAY("light_gray"),
    GRAY("gray"),
    PINK("pink"),
    LIME("lime"),
    YELLOW("yellow"),
    LIGHT_BLUE("light_blue"),
    MAGENTA("magenta"),
    ORANGE("orange"),
    WHITE("white");

    companion object {
        val ALL = values()
        val NAMES = ALL.map { it.asString }
    }
}